R/shinywrappers.R

```diff
@@ -176,12 +202,44 @@ print.shiny.render.function <- function(x, ...) {
 #'   )
 #' }
 #'
+#' # For better legacy support, consider using installExprFunction() over quoToFunction()
+#' renderTripleLegacy <- function(expr, env = parent.frame(), quoted = FALSE) {
+#'   func <- installExprFunction(expr, "func", env, quoted)
+#'
+#'   createRenderFunction(
+#'     func,
+#'     transform = function(value, session, name, ...) {
+#'       paste(rep(value, 3), collapse=", ")
+#'     },
+#'     outputFunc = textOutput
+#'   )
+#' }
+#'
 #' # Test render function from the console
-#' a <- 1
-#' r <- renderTriple({ a * 10 })
-#' a <- 2
-#' r()
-#' # [1] "20, 20, 20"
+#' reactiveConsole(TRUE)
+#'
+#' r <- reactiveVal()
+#' r("basic")
+#' renderTriple(r())()
+#' #> [1] "basic, basic, basic"
+#'
+#' # User can supply quoted code via rlang::quo()
+#' r2 <- rlang::quo({ r("rlang"); r() })
+#' rlang::inject(renderTriple(!!r2))()
+#' #> [1] "rlang, rlang, rlang"
+#'
+#' # Supplying quoted code without rlang::quo() requires installExprFunction()
+#' renderTripleLegacy(quote({ r("legacy"); r() }), quoted = TRUE)()
+#' #> [1] "legacy, legacy, legacy"
+#'
+#' # The legacy approach also supports with quosures (env is ignored in this case)
+#' r3 <- rlang::quo({ r("legacy-rlang"); r() })
+#' renderTripleLegacy(r3, quoted = TRUE)()
+#' #> [1] "legacy-rlang, legacy-rlang, legacy-rlang"
```

Comment:
```suggestion
# User can supply quoted code via rlang::quo(). Note that evaluation of the
# expression happens when r2() is invoked, not when r2 is created.
q <- rlang::quo({ v() })
r2 <- rlang::inject(renderTriple(!!q))
v("rlang")
r2()
#> [1] "rlang, rlang, rlang"

# Supplying quoted code without rlang::quo() requires installExprFunction()
expr <- quote({ v() })
r3 <- renderTripleLegacy(expr, quoted = TRUE)
v("legacy")
r3()
#> [1] "legacy, legacy, legacy"

# The legacy approach also supports with quosures (env is ignored in this case)
q <- rlang::quo({ v() })
r4 <- renderTripleLegacy(q, quoted = TRUE)
v("legacy-rlang")
r4()
#> [1] "legacy-rlang, legacy-rlang, legacy-rlang"
```

Reply:
This also makes me think -- we need to explain somewhere why it's better to create a render function **without** `env` and `quoted` and have people use `inject()`, rather than just keep using `quoted` when they have quosures.